src/js/core/methods/helpers/cardanoCertificate.js

```diff
@@ -122,38 +125,62 @@ const validatePoolParameters = (poolParameters: CardanoPoolParameters) => {
     }
 };
 
+export type CertificateWithPoolOwnersAndRelays = {
+    certificate: CardanoTxCertificateType,
+    poolOwners: CardanoPoolOwnerProto[],
+    poolRelays: CardanoPoolRelayProto[],
+};
+
+export type PoolParametersWithOwnersAndRelays = {
+    poolParameters?: CardanoPoolParametersType,
+    poolOwners: CardanoPoolOwnerProto[],
+    poolRelays: CardanoPoolRelayProto[],
+};
+
 const transformPoolParameters = (
-    poolParameters: CardanoPoolParameters,
-): CardanoPoolParametersType => {
+    poolParameters?: CardanoPoolParameters,
+): PoolParametersWithOwnersAndRelays => {
+    if (!poolParameters) {
+        return { poolParameters: undefined, poolOwners: [], poolRelays: [] };
+    }
+
     validatePoolParameters(poolParameters);
 
     return {
-        pool_id: poolParameters.poolId,
-        vrf_key_hash: poolParameters.vrfKeyHash,
-        pledge: poolParameters.pledge,
-        cost: poolParameters.cost,
-        margin_numerator: poolParameters.margin.numerator,
-        margin_denominator: poolParameters.margin.denominator,
-        reward_account: poolParameters.rewardAccount,
-        owners: poolParameters.owners.map(owner => ({
+        poolParameters: {
+            pool_id: poolParameters.poolId,
+            vrf_key_hash: poolParameters.vrfKeyHash,
+            pledge: poolParameters.pledge,
+            cost: poolParameters.cost,
+            margin_numerator: poolParameters.margin.numerator,
+            margin_denominator: poolParameters.margin.denominator,
+            reward_account: poolParameters.rewardAccount,
+            owners: [],
```

Comment:
does it make sense to keep owners and relays here if they are attached aside?Shouldn't there be the counts of owners and relays so Trezor knows how many to expect?

Reply:
ah I see, the counts are below - perhaps I'd clarify the empty arrays are here just for wire compatibility with the legacy, non-streamed Cardano tx signing